src/storage/accessors/SparqlDataAccessor.ts

```diff
@@ -137,6 +137,21 @@ export class SparqlDataAccessor implements DataAccessor {
     return this.sendSparqlUpdate(this.sparqlInsert(name, metadata, parent, triples));
   }
 
+  public async writeMetadata(identifier: ResourceIdentifier, metadata: RepresentationMetadata): Promise<void> {
+    const { name, parent } = this.getRelatedNames(identifier);
+    const triples = metadata.quads();
+
+    const def = defaultGraph();
+    if (triples.some((triple): boolean => !def.equals(triple.graph))) {
+      throw new NotImplementedHttpError('Only triples in the default graph are supported.');
+    }
+
+    // Not relevant since all content is triples
+    metadata.removeAll(CONTENT_TYPE);
+
+    return this.sendSparqlUpdate(this.sparqlInsert(name, metadata, parent));
```

Comment:
You can't reuse the `sparqlInsert` function here. That one completely deletes all triples associated with `name` first. So this call will remove all data in the target resource.

A new query will have to be generated for this that only modifies metadata, which will probably be a smaller part of the `sparqlInsert` call.